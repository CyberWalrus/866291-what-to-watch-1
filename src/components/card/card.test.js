import React from "react";
import renderer from "react-test-renderer";
import Card from "./card.jsx";

it(`Card correctly renders after relaunch`, () => {
  const clickFn = jest.fn();
  const tree = renderer
    .create(
        <Card
          id={1}
          genre="comedy"
          title="What We Do in the Shadows"
          src="img/what-we-do-in-the-shadows.jpg"
          onClick={clickFn}
          onMouseEnter={clickFn}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
