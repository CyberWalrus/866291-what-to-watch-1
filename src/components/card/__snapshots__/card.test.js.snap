// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card correctly renders after relaunch 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[MockFunction]}
  onMouseLeave={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
      controls={false}
      height="175"
      loop={true}
      muted={true}
      poster="https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="280"
    />
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        Shutter Island
      </a>
    </h3>
  </div>
</article>
`;
